testfixtures/logcapture.py

```diff
@@ -73,12 +95,51 @@ def atexit(cls):
                 '%s' % ('\n'.join((str(i.names) for i in cls.instances)))
                 )
 
+    def __len__(self):
+        return len(self.records)
+
+    def __getitem__(self, key):
+        if callable(self.attributes):
+            return self.attributes(self.records[key])
+        else:
+            result = tuple(self._actual_row(self.records[key]))
+            if len(result) == 1:
+                return result[0]
+            else:
+                return result
+
+    def __contains__(self, what):
+        for i in range(len(self)):
+            item = self[i]
+            if what == item:
+                self.asserted[i] = True
+                return True
+        return False
+
     def clear(self):
-        "Clear any entries that have been captured."
+        """Clear any entries that have been captured."""
         self.records = []
+        self.asserted = []
+
+    def mark_errors_as_asserted(self):
+        """ Mark all captured ERROR logs as asserted,
+         use this when doing the assertion externally """
+        for i in range(len(self.records)):
+            self.asserted[i] = True
+
+    def assert_no_error(self):  # (...) -> None
+        """ raise  AssertionError if there are un-asserted ERROR logs in this capture """
+        i = 0
+        for record in self.records:
+            if record.levelno == logging.ERROR and not self.asserted[i]:
+                raise AssertionError((
+                    'Not asserted ERROR log: %s'
+                ) % (pformat(self[i])))
```

Comment:
I think we want an exception that highlights *all* the messages not asserted, rather than just the first one.